src/Orleans.Core/Streams/PubSub/ImplicitStreamPubSub.cs

```diff
@@ -97,6 +97,11 @@ internal bool IsImplicitSubscriber(IAddressable addressable, StreamId streamId)
             return implicitTable.IsImplicitSubscriber(GrainExtensions.GetGrainId(addressable), streamId);
         }
 
+        internal bool IsImplicitSubscribeEligibleStreamNameSpace(StreamId streamId)
```

Comment:
I'm concerned this constraint is duplicated in multiple places.  Here, ImplicitStreamPubSub.RegisterProducer, and ImplicitStreamSubscriberTable.GetImplicitSubscribers.  I think we need a single authoritative place which defines this constraint (probably the ImplicitStreamSubscriberTable) and all other places reference to that logic.